lib/build/web_components.dart

```diff
@@ -25,7 +25,7 @@ Asset generateWebComponentsBootstrap(Resolver resolver, Transform transform,
     dom.Document document, AssetId scriptId, AssetId newScriptId,
     {List<InitializerPlugin> extraPlugins: const []}) {
   var htmlImportRecorder = new HtmlImportAnnotationRecorder();
-  var plugins = [htmlImportRecorder]..addAll(extraPlugins);
+  var plugins = [htmlImportRecorder as InitializerPlugin]..addAll(extraPlugins);
```

Comment:
This shouldn't be necessary afaik, try specifying `<InitializerPlugin>[htmlImportRecorder]` instead? I think the actual issue here was it was creating a `List<HtmlImportAnnotationRecorder>` and that was failing when you tried to add the `List<InitializerPlugin>` to it.

cc @munificent @vsmenon , not sure if this thing comes up often for strong mode or its more of an outlier?